### ISSUE TITLE
Non Seed containers in cassandra stack are not able to connet to seed node after upgrading to rancher V1.2

### ISSUE THREAD

**Issue by aaditi30** [rancher/rancher#6980]
my docker compose worked properly on rancher v 1.1.0. We upgraed to rancher v1.2.0 yesterday. After resolving issues with docker compose version 2 and updating docker compose. I was able to start cassandra seed node. Other non seed containers come up but they are not able to join seed node.

At first i though that its a security group issue but even after opening entire security group i keep getting same problem. Other containers come up and become unhealthy and this keeps going on.

Also we observe following errors in container logs after porting them to host:
Dec  8 05:21:20 ip-10-6-31-127 kernel: [79217.080016] device veth7b5bb1e5 entered promiscuous mode
Dec  8 05:21:20 ip-10-6-31-127 kernel: [79217.852089] docker0: port 6(veth7b5bb1e5) entered forwarding state
Dec  8 05:21:20 ip-10-6-31-127 kernel: [79217.852102] docker0: port 6(veth7b5bb1e5) entered forwarding state
Dec  8 05:21:33 ip-10-6-31-127 kernel: [79230.592322] docker0: port 6(veth7b5bb1e5) entered disabled state
Dec  8 05:21:33 ip-10-6-31-127 kernel: [79230.592852] device veth7b5bb1e5 left promiscuous mode
Dec  8 05:21:33 ip-10-6-31-127 kernel: [79230.592861] docker0: port 6(veth7b5bb1e5) entered disabled state
Dec  8 05:21:37 ip-10-6-31-127 kernel: [79234.258931] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Dec  8 05:21:37 ip-10-6-31-127 kernel: [79234.312887] device vetheccbd4a8 entered promiscuous mode
Dec  8 05:21:37 ip-10-6-31-127 kernel: [79234.314478] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Dec  8 05:21:38 ip-10-6-31-127 kernel: [79235.256111] docker0: port 6(vetheccbd4a8) entered forwarding state
Dec  8 05:21:38 ip-10-6-31-127 kernel: [79235.256123] docker0: port 6(vetheccbd4a8) entered forwarding state
Dec  8 05:21:45 ip-10-6-31-127 kernel: [79242.640397] docker0: port 6(vetheccbd4a8) entered disabled state
Dec  8 05:21:45 ip-10-6-31-127 kernel: [79242.640900] device vetheccbd4a8 left promiscuous mode
Dec  8 05:21:45 ip-10-6-31-127 kernel: [79242.640909] docker0: port 6(vetheccbd4a8) entered disabled state
Dec  8 05:21:47 ip-10-6-31-127 kernel: [79244.901179] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Dec  8 05:21:47 ip-10-6-31-127 kernel: [79244.934082] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Dec  8 05:21:47 ip-10-6-31-127 kernel: [79244.934445] device vethd30cd743 entered promiscuous mode
Dec  8 05:21:48 ip-10-6-31-127 kernel: [79245.900162] docker0: port 6(vethd30cd743) entered forwarding state
Dec  8 05:21:48 ip-10-6-31-127 kernel: [79245.900175] docker0: port 6(vethd30cd743) entered forwarding state
Dec  8 05:22:01 ip-10-6-31-127 kernel: [79258.092470] docker0: port 6(vethd30cd743) entered disabled state
Dec  8 05:22:01 ip-10-6-31-127 kernel: [79258.094337] device vethd30cd743 left promiscuous mode
Dec  8 05:22:01 ip-10-6-31-127 kernel: [79258.094346] docker0: port 6(vethd30cd743) entered disabled state
Dec  8 05:22:03 ip-10-6-31-127 kernel: [79260.578402] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Dec  8 05:22:03 ip-10-6-31-127 kernel: [79260.864237] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Dec  8 05:22:03 ip-10-6-31-127 kernel: [79260.880484] device vethe6f37788 entered promiscuous mode
Dec  8 05:22:04 ip-10-6-31-127 kernel: [79261.576102] docker0: port 6(vethe6f37788) entered forwarding state
Dec  8 05:22:04 ip-10-6-31-127 kernel: [79261.576113] docker0: port 6(vethe6f37788) entered forwarding state
Dec  8 05:22:14 ip-10-6-31-127 kernel: [79271.784339] docker0: port 6(vethe6f37788) entered disabled state
Dec  8 05:22:14 ip-10-6-31-127 kernel: [79271.784853] device vethe6f37788 left promiscuous mode
Dec  8 05:22:14 ip-10-6-31-127 kernel: [79271.784862] docker0: port 6(vethe6f37788) entered disabled state
Dec  8 05:22:17 ip-10-6-31-127 kernel: [79274.581620] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Dec  8 05:22:17 ip-10-6-31-127 kernel: [79274.700162] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Dec  8 05:22:17 ip-10-6-31-127 kernel: [79274.720321] device veth560e7c60 entered promiscuous mode
Dec  8 05:22:18 ip-10-6-31-127 kernel: [79275.580088] docker0: port 6(veth560e7c60) entered forwarding state
Dec  8 05:22:18 ip-10-6-31-127 kernel: [79275.580099] docker0: port 6(veth560e7c60) entered forwarding state
Dec  8 05:22:27 ip-10-6-31-127 kernel: [79284.956363] docker0: port 6(veth560e7c60) entered disabled state
Dec  8 05:22:27 ip-10-6-31-127 kernel: [79284.956867] device veth560e7c60 left promiscuous mode
Dec  8 05:22:27 ip-10-6-31-127 kernel: [79284.956876] docker0: port 6(veth560e7c60) entered disabled state
Dec  8 05:22:31 ip-10-6-31-127 kernel: [79288.024037] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Dec  8 05:22:31 ip-10-6-31-127 kernel: [79288.292162] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Dec  8 05:22:31 ip-10-6-31-127 kernel: [79288.292226] device vethc9394542 entered promiscuous mode
Dec  8 05:22:32 ip-10-6-31-127 kernel: [79289.020090] docker0: port 6(vethc9394542) entered forwarding state
Dec  8 05:22:32 ip-10-6-31-127 kernel: [79289.020102] docker0: port 6(vethc9394542) entered forwarding state
Dec  8 05:22:41 ip-10-6-31-127 kernel: [79298.108489] docker0: port 6(vethc9394542) entered disabled state
Dec  8 05:22:41 ip-10-6-31-127 kernel: [79298.110363] device vethc9394542 left promiscuous mode
Dec  8 05:22:41 ip-10-6-31-127 kernel: [79298.110372] docker0: port 6(vethc9394542) entered disabled state
Dec  8 05:22:45 ip-10-6-31-127 kernel: [79302.057117] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Dec  8 05:22:45 ip-10-6-31-127 kernel: [79302.188182] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Dec  8 05:22:45 ip-10-6-31-127 kernel: [79302.243712] device vethb96bf018 entered promiscuous mode
Dec  8 05:22:46 ip-10-6-31-127 kernel: [79302.996089] docker0: port 6(vethb96bf018) entered forwarding state
Dec  8 05:22:46 ip-10-6-31-127 kernel: [79302.996101] docker0: port 6(vethb96bf018) entered forwarding state
Dec  8 05:22:50 ip-10-6-31-127 kernel: [79307.900364] docker0: port 6(vethb96bf018) entered disabled state
Dec  8 05:22:50 ip-10-6-31-127 kernel: [79307.900835] device vethb96bf018 left promiscuous mode
Dec  8 05:22:50 ip-10-6-31-127 kernel: [79307.900842] docker0: port 6(vethb96bf018) entered disabled state
Dec  8 05:22:54 ip-10-6-31-127 kernel: [79311.188552] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Dec  8 05:22:54 ip-10-6-31-127 kernel: [79311.235064] device veth6bf44fc1 entered promiscuous mode
Dec  8 05:22:54 ip-10-6-31-127 kernel: [79311.244098] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Dec  8 05:22:55 ip-10-6-31-127 kernel: [79312.192086] docker0: port 6(veth6bf44fc1) entered forwarding state
Dec  8 05:22:55 ip-10-6-31-127 kernel: [79312.192099] docker0: port 6(veth6bf44fc1) entered forwarding state
Dec  8 05:23:04 ip-10-6-31-127 kernel: [79321.360355] docker0: port 6(veth6bf44fc1) entered disabled state
Dec  8 05:23:04 ip-10-6-31-127 kernel: [79321.362222] device veth6bf44fc1 left promiscuous mode
Dec  8 05:23:04 ip-10-6-31-127 kernel: [79321.362230] docker0: port 6(veth6bf44fc1) entered disabled state
Dec  8 05:23:06 ip-10-6-31-127 kernel: [79323.884230] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Dec  8 05:23:06 ip-10-6-31-127 kernel: [79323.917546] device veth6d73ae90 entered promiscuous mode
Dec  8 05:23:06 ip-10-6-31-127 kernel: [79323.918376] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Dec  8 05:23:07 ip-10-6-31-127 kernel: [79324.884086] docker0: port 6(veth6d73ae90) entered forwarding state
Dec  8 05:23:07 ip-10-6-31-127 kernel: [79324.884098] docker0: port 6(veth6d73ae90) entered forwarding state
Dec  8 05:23:11 ip-10-6-31-127 dhclient: DHCPREQUEST of 10.6.31.127 on eth0 to 10.6.31.1 port 67 (xid=0x65007980)
Dec  8 05:23:11 ip-10-6-31-127 dhclient: DHCPACK of 10.6.31.127 from 10.6.31.1
Dec  8 05:23:11 ip-10-6-31-127 dhclient: bound to 10.6.31.127 -- renewal in 1351 seconds.
Dec  8 05:23:15 ip-10-6-31-127 kernel: [79332.972368] docker0: port 6(veth6d73ae90) entered disabled state
Dec  8 05:23:15 ip-10-6-31-127 kernel: [79332.974233] device veth6d73ae90 left promiscuous mode
Dec  8 05:23:15 ip-10-6-31-127 kernel: [79332.974242] docker0: port 6(veth6d73ae90) entered disabled state
Dec  8 05:23:19 ip-10-6-31-127 kernel: [79336.174816] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Dec  8 05:23:19 ip-10-6-31-127 kernel: [79336.312153] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Dec  8 05:23:19 ip-10-6-31-127 kernel: [79336.320365] device vethbca3558d entered promiscuous mode
Dec  8 05:23:20 ip-10-6-31-127 kernel: [79337.136086] docker0: port 6(vethbca3558d) entered forwarding state
Dec  8 05:23:20 ip-10-6-31-127 kernel: [79337.136099] docker0: port 6(vethbca3558d) entered forwarding state
Dec  8 05:23:31 ip-10-6-31-127 kernel: [79348.196520] docker0: port 6(vethbca3558d) entered disabled state
Dec  8 05:23:31 ip-10-6-31-127 kernel: [79348.198459] device vethbca3558d left promiscuous mode
Dec  8 05:23:31 ip-10-6-31-127 kernel: [79348.198468] docker0: port 6(vethbca3558d) entered disabled state
Dec  8 05:23:33 ip-10-6-31-127 kernel: [79350.897217] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Dec  8 05:23:34 ip-10-6-31-127 kernel: [79351.000403] device veth2119a971 entered promiscuous mode
Dec  8 05:23:34 ip-10-6-31-127 kernel: [79351.068167] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Dec  8 05:23:34 ip-10-6-31-127 kernel: [79351.896113] docker0: port 6(veth2119a971) entered forwarding state
Dec  8 05:23:34 ip-10-6-31-127 kernel: [79351.896125] docker0: port 6(veth2119a971) entered forwarding state
Dec  8 05:23:44 ip-10-6-31-127 kernel: [79361.480381] docker0: port 6(veth2119a971) entered disabled state
Dec  8 05:23:44 ip-10-6-31-127 kernel: [79361.480850] device veth2119a971 left promiscuous mode
Dec  8 05:23:44 ip-10-6-31-127 kernel: [79361.480857] docker0: port 6(veth2119a971) entered disabled state
Dec  8 05:23:47 ip-10-6-31-127 kernel: [79364.839424] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Dec  8 05:23:48 ip-10-6-31-127 kernel: [79365.004186] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Dec  8 05:23:48 ip-10-6-31-127 kernel: [79365.036308] device vethd2e624e8 entered promiscuous mode
Dec  8 05:23:48 ip-10-6-31-127 kernel: [79365.836089] docker0: port 6(vethd2e624e8) entered forwarding state
Dec  8 05:23:48 ip-10-6-31-127 kernel: [79365.836102] docker0: port 6(vethd2e624e8) entered forwarding state

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v1.2.0`  UI: `v1.2.34` |
|Access|`localauth` `admin`|
|Orchestration|`docker-compose`|
|Route|`stack.index`|

**Comment by aaditi30** [comment]
If i try to create a single container keep getting this error:
Failed to allocate instance [container:1i293]: Bad instance [container:1i293] in state [stopping]: Scheduling failed: needs host with label name: vm2

I have labeled VM's properly so not having the vm lable is not the issue

**Comment by tfiduccia** [comment]
@aaditi30 - for the failed Container, if you look at the host page in UI, you have a host that has this on it?
![image](https://cloud.githubusercontent.com/assets/11514927/21022310/6b89b5c6-bd39-11e6-948d-07b2bf85ae2d.png)

**Comment by aaditi30** [comment]
Hi tfiduccia,
Thanks for the response. I have set  label name for each host and i do have a host with vm name VM2.

seed cassandra container comes up. Second container has problems connecting to the first container and shows the logs that i have attached above. 

In current docker compose i have used link to connect secondary container to primary/ seed container. Is there any change in Rancher V2.0 for networking.

**Comment by tfiduccia** [comment]
@aaditi30 - the reason why your container is failing to find the host is because the label you have is uppercase and our new scheduler is no longer case insensitive.  Until we can resolve this issue can you change your host label to be all lower case?

UI is lowercasing it, even if you enter uppercase.  The scheduler is supposed to be (and used to be) case insensitive.

**Comment by cjellick** [comment]
We investigated and it is not that the scheduler is now case sensitive. There is a weird condition where going from upper to lower case when editing a label causes us to not create an entry in the database.

**Comment by cjellick** [comment]
@tfiduccia @deniseschannon given that this is not caused by any changes in the scheduler and that it is not as sever as originally thought (we originally thought all label scheduling was now case sensitive), can we either assign to someone and/or move out of 1.2.1?

**Comment by alena1108** [comment]
Steps to test:

1) Create host
2) Add label with key=foo, value=bar. As a result, label is set on the host object, and internally, in the labels table. You can fetch this label using this query:

```
select label.key, label.value from label where id in (select label_id from host_label_map where host_id=1 and removed is null);
```
3) Modify the label so the key=Foo, value=bar. Execute mysql query again, make sure the result didn't change. Before the fix, the foo=bar label used to be removed from the list.

**Comment by sangeethah** [comment]
Labels do not get lost when case of the name/value is edited and saved.
There was a relating  bug that was also addressed as part of - https://github.com/rancher/rancher/issues/7874

@aaditi30 , the fix for the problem you are encountering will be available in 1.5 release.
